agent/lib/kontena/helpers/weave_helper.rb

```diff
@@ -5,11 +5,22 @@ module Kontena
   module Helpers
     module WeaveHelper
 
-      # @return [Boolean]
-      def weave_running?
-        weave = Docker::Container.get('weave') rescue nil
-        return false if weave.nil?
-        weave.info['State']['Running'] == true
+      def network_adapter
+        Celluloid::Actor[:network_adapter]
+      end
+
+      def wait_weave_running?(timeout = 10, &block)
```

Comment:
No idea really, I assume Docker uses some form of exponential backoff, so it depends on how slow the agent is to get weave running, in terms of how many times Docker attempts to restart the `kontena-etcd` and `kontena-ipam` containers before they actually come up :/

I think it can sometimes take a while... I'd maybe consider something like 300s just to be on the safe side, while still avoiding the worst of the current "just silently wait for infinity"